src/utils/configUtils.ts

```diff
@@ -40,7 +41,38 @@ export async function loadRunConfig(workspaceFolder: WorkspaceFolder): Promise<I
             return defaultConfigs[0];
         }
     }
-    return await selectQuickPick(configItems, workspaceFolder);
+
+    const candidateConfigItems: IExecutionConfig[] = filterCandidateConfigItems(configItems, testItems);
+    return await selectQuickPick(candidateConfigItems, workspaceFolder);
+}
+
+function filterCandidateConfigItems(configItems: IExecutionConfig[], testItems: TestItem[]): IExecutionConfig[] {
+    return configItems.filter((config: IExecutionConfig) => {
+        const whenClause: string | undefined = config.when?.trim();
+
+        if (whenClause) {
+            const context: WhenClauseEvaluationContext = new WhenClauseEvaluationContext(whenClause);
+
+            try {
+                return checkTestItems(testItems, context);
+            } catch (e) {
+                if (e instanceof Error)
+                    window.showWarningMessage(e.message);
```

Comment:
Usually we should be careful about using the notifications. Because it will grab the user's attention. See: https://code.visualstudio.com/api/ux-guidelines/notifications

Here, I would recommend failing quietly for those configurations have a wrong when clause. (in other word, ignore the when clause if it's not written correctly).

If we want to let user know the when clause is wrong, we can use things like diagnostics: https://github.com/microsoft/vscode-extension-samples/tree/main/diagnostic-related-information-sample.

But it's out of the scope of this PR, so far just ignore the when clause will be good enough to me :)

Reply:
Fair enough - assume this applies to https://github.com/microsoft/vscode-java-test/blob/b231049b1bd33950a32c3d55c7d89c365f729574/src/utils/configUtils.ts#L208 too? I'll leave the try/catch block there so we can still throw errors from `evaluate` and return the configuration as a candidate.

Reply:
https://github.com/microsoft/vscode-java-test/pull/1638/commits/df5da6b4d71d294d8496608ab39d432a3ab06be4